docs/instrument-guide.md

```diff
@@ -0,0 +1,111 @@
+# Adding a New Instrumentation Hook
+
+This guide outlines the workflow for adding compile-time instrumentation for a third-party library.
+
+The process consists of three main steps:
+
+1. **Define Rules**: Create a YAML file to match the target package and function.
+2. **Implement Hooks**: Write the `Before` and `After` hook functions in Go.
+3. **Verify**: Add tests to ensure the instrumentation works as expected.
+
+---
+
+## 1. Define Rules
+
+Rules are defined in YAML format and stored in `tool/data/`. These files tell the `otel` which functions to instrument.
+
+Create a new file `tool/data/<library-name>.yaml`. Below is an example configuration for instrumenting a function `NewServer`:
+
+```yaml
+inject_to_grpc_newserver:
+  target: google.golang.org/grpc
+  version: v1.63.0,v1.70.0
+  func: NewServer
+  before: BeforeNewServer
+  after: AfterNewServer
+  path: github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/grpc/server
+```
+
+* `target`: Import path of the package to instrument.
+* `version`: Version range to match.
+* `func`: Name of the function to hook.
```

Comment:
What is the behavior if func does not exist?

Reply:
Changed to "Name of the function to hook. (Required)"